akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -138,6 +138,26 @@ final class Source[+Out, +Mat](
   // FIXME: Out => Unit should stay, right??
   def runForeach(f: Out ⇒ Unit)(implicit materializer: Materializer): Future[Done] = runWith(Sink.foreach(f))
 
+  /**
+   * Shortcut for running this `Source` to retrieve the first value received.
+   * If the stream completes before signaling at least a single element, the Future will be failed with a [[NoSuchElementException]].
+   * If the stream signals an error errors before signaling at least a single element, the Future will be failed with the streams exception.
+   *
+   * See also [[runHeadOption]].
+   */
+  def runHead(implicit materializer: Materializer): Future[Out] =
```

Comment:
why does `head` deserve its own `run`? Why isn't `runWith(Sink.head)` convenient enough?

Reply:
like `runWith(Sink.foreach(f))`, I think `runWith(Sink.head)` is so common so, with this method, I could write all my code in the domain of `Source[T]` not both.

Reply:
Common? Perhaps in tests but I can’t imagine that it is very interersting for real streams.

Reply:
When I am using this, I see the *Stream* as a *Stream* of element `0 to 1`, for this kind of, it's common :)

Reply:
That does not really address the question of it being really common and what we should show people. People *should* learn about Sinks and Sources early on. It's the key feature and mindset-thing of akka streams